<launch>
    <arg name="with_logging_manager" default="false"/>
    <arg name="path_visualisation" default="false"/>
    <arg name="online_construction" default="true"/>
    <arg name="tracker_topic" default="/people_tracker/positions"/>
    <arg name="ubd_topic" default="/upper_body_detector/bounding_box_centres"/>
    <arg name="collection_name" default="people_trajectory"/>
    <arg name="logging_manager_topic" default="/logging_manager/log_stamped"/>
    <arg name="map_info" default=""/>
    <arg name="max_trajectories" default="10000"/>

    <arg name="machine" default="localhost" />
    <arg name="user" default="" />

    <machine name="$(arg machine)" address="$(arg machine)" env-loader="$(optenv ROS_ENV_LOADER )" user="$(arg user)" default="true"/>

    <node pkg="human_trajectory" type="trajectory_publisher.py"
name="$(arg collection_name)" output="screen" respawn="true">
        <param name="with_logging_manager" value="$(arg with_logging_manager)" type="bool"/>
        <param name="path_visualisation" value="$(arg path_visualisation)" type="bool"/>
        <param name="online_construction" value="$(arg online_construction)" type="bool"/>
        <param name="tracker_topic" value="$(arg tracker_topic)" type="string"/>
        <param name="ubd_topic" value="$(arg ubd_topic)" type="string"/>
        <param name="collection_name" value="$(arg collection_name)" type="string"/>
        <param name="logging_manager_topic" value="$(arg logging_manager_topic)" type="string"/>
        <param name="map_info" value="$(arg map_info)" type="string"/>
        <param name="max_trajectories" value="$(arg max_trajectories)" type="int"/>
    </node>
</launch>
